crypto/fipsmodule/dh/check.c

```diff
@@ -60,6 +60,7 @@
 
 #include "internal.h"
 
+#define OPENSSL_DH_CHECK_MAX_MODULUS_BITS  32768
```

Comment:
Is there a reason that `32768` was chosen?  Why not smaller? Why not larger?

The bulletin you linked only mentions 10,000:
```
Trying to use
a very large modulus is slow and OpenSSL will not normally use a modulus which
is over 10,000 bits in length.
```

Reply:
The 10,000 bits is the size OpenSSL limits for generation currently, but they chose to use a larger number for checking, as keys not generated by OpenSSL may have been potentially larger.

Reply:
>The DH_check() function checks numerous aspects of the key or parameters
that have been supplied. Some of those checks use the supplied modulus
value even if it is excessively large.
>
>There is already a maximum DH modulus size (10,000 bits) over which
OpenSSL will not generate or derive keys. DH_check() will however still
perform various tests for validity on such a large modulus. We introduce a
new maximum (32,768) over which DH_check() will just fail.
>
>An application that calls DH_check() and supplies a key or parameters
obtained from an untrusted source could be vulnerable to a Denial of
Service attack.
>
>The function DH_check() is itself called by a number of other OpenSSL
functions. An application calling any of those other functions may
similarly be affected. The other functions affected by this are
DH_check_ex() and EVP_PKEY_param_check().

The original openssl commit message details explains better.